### ISSUE TITLE
[HOWTO] Turn off auto formatting code in c/c++

### ISSUE THREAD

**Issue by chengyi** [hlissner/doom-emacs#2092]
**What I want to achieve**
when I write c/c++ code, the code is auto formatted like below, How to turn off this.

![Peek 2019-11-21 16-15](https://user-images.githubusercontent.com/6308364/69321106-70bb7480-0c7d-11ea-9ab1-4724943fbca5.gif)
 
I already test without my own config. @hlissner 

**System information**
<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      8月 30, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop 758df8aef 2019-11-20 10:30:46 +0800
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 4.19.84-1-MANJARO #1 SMP PREEMPT Wed Nov 13 00:07:37 UTC 2019 x86_64
        path       (~/.local/bin /usr/local/bin /usr/bin /bin /usr/local/sbin /opt/android-ndk /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.3/x86_64-pc-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion company (ivy +fuzzy) prow-company :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) vc-gutter vi-tilde-fringe window-select workspaces prow-highlight prow-hydra prow-treemacs :editor file-templates fold snippets prow-edit :emacs vc :term vterm :tools eval flycheck (lookup +docsets) lsp magit gtags :lang (cc +lsp) prow-c-c++ emacs-lisp (org +dragndrop +ipython +pandoc +present) (python +lsp) (rust +lsp) (sh +zsh) prow-markdown :email mu4e :config (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>

**Comment by myrgy** [comment]
Hi @chengyi ,
seence you have LSP I think that's formatting from comes from it.
try to disable it:
 `(setq lsp-enable-on-type-formatting nil)`  (see https://github.com/emacs-lsp/lsp-mode)
in your private `config.el`

**Comment by chengyi** [comment]
@myrgy I set this var using
``` emacs-lisp
(use-package-hook! lsp-mode
  :post-config
  (setq lsp-enable-on-type-formatting nil))
```
and this code take efftecs.
But the issue still unresolved.

**Comment by hlissner** [comment]
I cannot reproduce this. It looks like an issue with your setting for `tab-width` or `c-basic-offset` being set to 4, when your code is using 2-spaces of indentation.newline-and-indent` (on `RET`) will reindent the new line, so this is correct behavior.

Change your `c-basic-offset` to the indentation you want, or use a program like editorconfig to set it externally for your project.

**Comment by chengyi** [comment]
@hlissner but how to avoid auto formatting?

**Comment by hlissner** [comment]
```elisp
(map! [remap newline] nil)
```

**Comment by chengyi** [comment]
@hlissner I use emacs keybinding, and I discover that even if I create a new c file with clean doom emacs without my private config, emacs still set c-basic-offset and tab-width to 2. it's strange.

**Comment by hlissner** [comment]
> I use emacs keybinding

Then remove the `:i` in my snippet.

> emacs still set c-basic-offset and tab-width to 2

Your guess is as good as mine. Maybe something in your private config?

**Comment by chengyi** [comment]
@hlissner no config in my private config, and I test this in clean doom emacs using sandbox `C-h E`

**Comment by hlissner** [comment]
I cannot reproduce this. In vanilla Emacs or Doom. Can you give exact steps to reproduce this?

**Comment by hlissner** [comment]
And please check the value of `c-basic-offset` and `tab-width` manually in your c/c++ buffer with `SPC h v VARNAME`. Those reports in messages indicate they've been changed, but something could still change those values later.

**Comment by chengyi** [comment]
@hlissner it seems that doom sandbox still load my private config even I press `C-c C-p`, in my private config, use the `aggressive-indent` package which lead to this strange behavior.

**Comment by hlissner** [comment]
> aggressive-indent

Ah, that makes sense. Understood, I'll close this issue then.

> it seems that doom sandbox still load my private config even I press C-c C-p

Judging from other issues you've filed, I suspect this is because you have a lot of private config in init.el, which the sandbox reads (with the assumption that anything you put in init.el is important enough to be loaded for the sandbox), likely with `use-package-hook!`. Users are expected to configure Doom from config.el. I'll look into a fix.

**Comment by hlissner** [comment]
> it seems that doom sandbox still load my private config even I press C-c C-p

Just pushed a fix for this in 441fc51.

**Comment by chengyi** [comment]
@hlissner excellent! :+1: :grin:
